diff --git a/networking_cisco/plugins/ml2/drivers/cisco/nexus/nexus_db_v2.py b/networking_cisco/plugins/ml2/drivers/cisco/nexus/nexus_db_v2.py
index 67f6a3d..0a63266 100644
--- a/networking_cisco/plugins/ml2/drivers/cisco/nexus/nexus_db_v2.py
+++ b/networking_cisco/plugins/ml2/drivers/cisco/nexus/nexus_db_v2.py
@@ -14,7 +14,9 @@
 #    under the License.
 #
 
+from oslo_db import exception as db_exc
 from oslo_log import log as logging
+from oslo_utils import excutils
 from random import shuffle
 import sqlalchemy as sa
 from sqlalchemy.orm import aliased
@@ -508,8 +510,18 @@ def add_host_mapping(host_id, nexus_ip, interface, ch_grp, is_static):
                   switch_ip=nexus_ip,
                   ch_grp=ch_grp,
                   is_static=is_static)
-    session.add(mapping)
-    session.flush()
+    try:
+        session.add(mapping)
+        session.flush()
+    except db_exc.DBDuplicateEntry:
+        with excutils.save_and_reraise_exception() as ctxt:
+            if is_static:
+                ctxt.reraise = False
+                LOG.debug("Duplicate static entry encountered "
+                          "host=%(host)s, if=%(if)s, ip=%(ip)s",
+                          {'host': host_id, 'if': interface,
+                           'ip': nexus_ip})
+
     return mapping
 
 
diff --git a/networking_cisco/plugins/ml2/drivers/cisco/nexus/nexus_models_v2.py b/networking_cisco/plugins/ml2/drivers/cisco/nexus/nexus_models_v2.py
index 050b2a6..c24babf 100644
--- a/networking_cisco/plugins/ml2/drivers/cisco/nexus/nexus_models_v2.py
+++ b/networking_cisco/plugins/ml2/drivers/cisco/nexus/nexus_models_v2.py
@@ -98,7 +98,8 @@ class NexusHostMapping(bc.model_base.BASEV2):
 
     __tablename__ = 'cisco_ml2_nexus_host_interface_mapping'
 
-    host_id = sa.Column(sa.String(255), nullable=False, index=True)
+    host_id = sa.Column(sa.String(255), nullable=False, primary_key=True,
+                        index=True)
     switch_ip = sa.Column(sa.String(255), nullable=False, primary_key=True,
                           index=True)
     if_id = sa.Column(sa.String(255), nullable=False, primary_key=True)
