--- overcloud_compute.pp       2016-01-27 00:19:48.257287463 -0800
+++ overcloud_compute.pp   2016-02-02 04:29:57.739932056 -0800
@@ -41,6 +41,7 @@
 nova_config {
   'DEFAULT/my_ip':                     value => $ipaddress;
   'DEFAULT/linuxnet_interface_driver': value => 'nova.network.linux_net.LinuxOVSInterfaceDriver';
+  'DEFAULT/resume_guests_state_on_host_boot': value => 'true';
 }

 $nova_enable_rbd_backend = hiera('nova_enable_rbd_backend', false)