diff --git a/networking_cisco/plugins/ml2/drivers/cisco/ucsm/config.py b/networking_cisco/plugins/ml2/drivers/cisco/ucsm/config.py
index 748a145..44b1137 100644
--- a/networking_cisco/plugins/ml2/drivers/cisco/ucsm/config.py
+++ b/networking_cisco/plugins/ml2/drivers/cisco/ucsm/config.py
@@ -89,7 +89,8 @@ def parse_ucsm_host_config():
             if not sep or not service_profile:
                 raise cfg.Error(_("UCS Mech Driver: Invalid Host Service "
                                   "Profile config: %s") % host)
-            key = (cfg.CONF.ml2_cisco_ucsm.ucsm_ip, hostname)
+            short_hostname = hostname.split('.')[0]
+            key = (cfg.CONF.ml2_cisco_ucsm.ucsm_ip, short_hostname)
             if '/' not in service_profile:
                 # Assuming the service profile is at the root level
                 # and the path is not specified. This option
@@ -104,8 +105,8 @@ def parse_ucsm_host_config():
                 sp_dict[key] = service_profile.strip()
 
             LOG.debug('Service Profile for %s is %s',
-                hostname, sp_dict.get(key))
-            host_dict[hostname] = cfg.CONF.ml2_cisco_ucsm.ucsm_ip
+                short_hostname, sp_dict.get(key))
+            host_dict[short_hostname] = cfg.CONF.ml2_cisco_ucsm.ucsm_ip
         return sp_dict, host_dict
 
 
